mlflow/transformers.py

```diff
@@ -426,7 +428,10 @@ def save_model(
         flavor_conf.update({_PROCESSOR_TYPE_KEY: _get_instance_type(processor)})
 
     # Save the pipeline object
-    built_pipeline.save_pretrained(save_directory=path.joinpath(_PIPELINE_BINARY_FILE_NAME))
+    built_pipeline.model.save_pretrained(
```

Comment:
Are we sure that we can stop saving the pipeline entirely in this way? I think there's a risk that the `max_shard_size` parameter may not be supported for all pipeline types (though not sure). What do we think about the following approach:
1. Try to `save_pretrained` on the model
2. If that doesn't work, just save the pipeline

Reply:
> max_shard_size parameter may not be supported for all pipeline types (though not sure)
Do you mean maybe not all "model" types support the `max_shard_size` param? It seems for me it's supported for any `transformers.PreTrainedModel` by reading its [doc](https://huggingface.co/docs/transformers/v4.29.1/en/main_classes/model#transformers.PreTrainedModel). 

But yeah I agree we can play safe. I can add the try catch thing. Do you have suggestion on which errors to catch? I'm not super familiar with HF errors, and don't want to accidentally swallow errors.

Reply:
If you think it's broadly supported I'm good with it - this comment mostly originates from me not understanding Huggingface pipelines' full level of flexibility

Reply:
For models who override `save_pretrained` and break its signature, I don't think mlflow transformers flavor is guaranteed to work, right?
I'll keep this unchanged (i.e. no try catch) for now then. We can revisit this if we later find the assumption is broken.